src/components/icon-button.tsx

```diff
@@ -0,0 +1,46 @@
+import { inject, observer } from "mobx-react";
+import * as React from "react";
+import { BaseComponent, IBaseProps } from "./base";
+import Button from "@material-ui/core/Button";
+
+import * as css from "./icon-button.scss";
+
+interface IProps extends IBaseProps {
+  icon: any;
+  highlightIcon: any;
+  buttonText: string;
+  onClick: any;
+  disabled: boolean;
+}
+
+interface IState {}
+
+@inject("stores")
+@observer
+export class IconButton extends BaseComponent<IProps, IState> {
```

Comment:
Since you extracted this component to a more generic one, I think it's a good candidate to be a "dumb component" that is not observing anything. Especially that it seems it's not using any data coming from the store.